### PR TITLE
Refactor  code that belongs to the Cart and History . Add  tests for the mvcapp.controller.CartControler getCarts method.

### PR DESCRIPTION
**Pull request by IhorKryviuk**

### CODE REVIEW

File: RESTserver/src/main/java/com/softserve/if072/restservice/service/CartService.java [hunk 1]
```diff
@@ -28,94 +28,105 @@
...
+    @Value("${cart.productExist}")
+    private String cartProductExist;
     @Value("${cart.notFound}")
     private String cartNotFound;
-    @Value("${cart.found}")
-    private String cartFound;
     @Value("${cart.SuccessfullyOperation}")
-    private String successfullyOperation;
+    private String cartSuccessfullyOperation;
     @Value("%{cart.foundProductId}")
```
Comment (IhorKryviuk) [inline RESTserver/src/main/java/com/softserve/if072/restservice/service/CartService.java:45]: I've already changed % to $

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CartController.java [hunk 2]
```diff
@@ -2,90 +2,65 @@
...
-    private RestTemplate template;
-    private UserService userService;
-
+public class CartController{
     @Autowired
-    public CartController(RestTemplate template, UserService userService) {
-        this.template = template;
-        this.userService = userService;
-    }
+    CartService cartService;
```
Comment (IhorKryviuk) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CartController.java:25]: I'll change to autowire via constructor wherever necessary in my code in the next commit.

Comment (kucherif) [reply MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/CartController.java:25]: Don't forget about access modifier

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/HistoryController.java [hunk 3]
```diff
@@ -1,66 +1,54 @@
...
-    private RestTemplate restTemplate;
-    private UserService userService;
-
+public class HistoryController{
     @Autowired
-    public HistoryController(RestTemplate restTemplate, UserService userService) {
-        this.restTemplate = restTemplate;
-        this.userService = userService;
-    }
+    HistoryService historyService;
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/HistoryController.java:25]: access modifier

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/CartService.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * @author Igor Kryviuk
+ */
+@Service
+public class CartService {
+    private static final Logger LOGGER = LogManager.getLogger();
+    @Autowired
+    private RestTemplate restTemplate;
+    @Autowired
+    private UserService userService;
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/CartService.java:27]: just to remember. use constructor for autowire

File: MVCapplication/src/test/com/softserve/if072/mvcapp/controller/CartControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+/**
+ * The CartControllerTest class is used to test CartController class methods
+ *
+ * @author Igor Kryviuk
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CartControllerTest {
```
Comment (kucherif) [inline MVCapplication/src/test/com/softserve/if072/mvcapp/controller/CartControllerTest.java:35]: 👍

**Pull request merged**